src/main/java/uk/nhs/tis/trainee/notifications/service/EventBroadcastService.java

```diff
@@ -0,0 +1,138 @@
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.time.Instant;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.bson.types.ObjectId;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+import software.amazon.awssdk.services.sns.model.PublishRequest.Builder;
+import software.amazon.awssdk.services.sns.model.SnsException;
+import uk.nhs.tis.trainee.notifications.config.EventNotificationProperties;
+import uk.nhs.tis.trainee.notifications.config.EventNotificationProperties.SnsRoute;
+import uk.nhs.tis.trainee.notifications.config.ObjectIdSerializerModule;
+import uk.nhs.tis.trainee.notifications.model.History;
+import uk.nhs.tis.trainee.notifications.model.NotificationStatus;
+
+/**
+ * A service for broadcasting form events to SNS.
+ */
+@Slf4j
+@Service
+public class EventBroadcastService {
+
+  public static final String MESSAGE_GROUP_ID_PREFIX = "notifications_event";
+
+  private final SnsClient snsClient;
+
+  private final EventNotificationProperties eventNotificationProperties;
+
+  EventBroadcastService(SnsClient snsClient,
+      EventNotificationProperties eventNotificationProperties) {
+    this.snsClient = snsClient;
+    this.eventNotificationProperties = eventNotificationProperties;
+  }
+
+  /**
+   * Publish a notification history event to SNS.
+   *
+   * @param history The history event to publish.
+   */
+  public void publishNotificationsEvent(History history) {
+
+    ObjectMapper objectMapper = new ObjectMapper()
+        .registerModule(new JavaTimeModule())
+        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+        .registerModule(new ObjectIdSerializerModule());
+
+    PublishRequest request = null;
+    SnsRoute snsTopic = eventNotificationProperties.notificationsEvent();
+
+    if (snsTopic != null && history != null) {
+      JsonNode eventJson = objectMapper.valueToTree(history);
+      request = buildSnsRequest(eventJson, snsTopic, history.id());
+    }
+
+    if (request != null) {
+      try {
+        snsClient.publish(request);
+        log.info("Broadcast event sent to SNS for notification event {}.", history.id());
+      } catch (SnsException e) {
+        String message = String.format(
+            "Failed to broadcast event to SNS topic '%s' for notification event '%s'",
+            snsTopic, history.id());
+        log.error(message, e);
+      }
+    }
+  }
+
+  /**
+   * Publish a blank record with NotificationStatus DELETED for a deleted history item.
+   *
+   * @param id The History id.
+   */
+  public void publishNotificationsDeleteEvent(ObjectId id) {
+    Instant sentAt = Instant.now();
+    History history = new History(id, null, null, null, null,
+        sentAt, null, NotificationStatus.DELETED, null, null);
```

Comment:
Is it ok to overload sentAt here, or should we use a new field?

Reply:
As all fields would be dummy when records is DELETED in the NDW, can't see a problem reusing sentAt for me. But I have added a field `statusDatetime` as the delete time of the Actions. I am thinking should I do similar to reuse the `completed` field for the time, or do you prefer having an extra field with same field name for consistency?